crates/wit-parser/src/embedder.rs

```diff
@@ -25,20 +25,24 @@ use wasmer_wit::{
 };
 use wasmer_wit::ToBytes;
 
-use std::path::PathBuf;
+use std::path::Path;
 
 /// Embed provided WIT to a Wasm file by path.
-pub fn embed_text_wit(in_wasm_path: PathBuf, out_wasm_path: PathBuf, wit: &str) -> Result<()> {
+pub fn embed_text_wit<I, O>(in_wasm_path: I, out_wasm_path: O, wit: &str) -> Result<()>
+where
+    I: AsRef<Path>,
+    O: AsRef<Path>,
```

Comment:
Do you really need this? `&Path` is OK in almost all cases

Reply:
It's an export func from a module, I think it worth here, because it would allow to pass `PathBuf` without copying.